### PR TITLE
Prepare for using Sallyport in SGX

### PR DESCRIPTION
**Pull request by lkatalin**
This series of commits:
- Passes a sallyport Block into the enclave and shim's handler
- Creates a new syscall proxy function that reads arguments out of the handler's Block, but otherwise works the same way as the previous function

Related to #614

### CODE REVIEW

**Comment by npmccallum** [comment]
@lkatalin Let's do just the first four commits in the first PR.

**Comment by lkatalin** [comment]
Sounds good @npmccallum , I'll re-push.

File: sallyport/src/lib.rs [hunk 1]
```diff
@@ -82,7 +82,8 @@ impl Request {
 #[repr(C)]
 #[derive(Copy, Clone, Default, PartialEq, Debug)]
 pub struct Reply {
-    ret: [Register<usize>; 2],
+    /// Return values from the syscall: rax:rdx
+    pub ret: [Register<usize>; 2],
```
Comment (npmccallum) [inline sallyport/src/lib.rs:86]: Definitely don't do this.

File: enarx-keep-sgx-shim/src/handler.rs [hunk 2]
```diff
@@ -103,6 +103,25 @@ impl<'a> Handler<'a> {
...
+            self.block.msg.req.arg[2].raw() as u64, // rdx
+            self.aex,
+            self.block.msg.req.arg[4].raw() as u64, // r8
+            self.block.msg.req.arg[5].raw() as u64, // r9
+            self.block.msg.req.arg[3].raw() as u64, // r10
+            self.block.msg.req.num.raw() as u64,    // rax
+            self.ctx,
+        );
+
+        self.block.msg.rep.ret[0] = ret.into();
```
Comment (npmccallum) [inline enarx-keep-sgx-shim/src/handler.rs:120]:
```suggestion
        self.block.msg.rep = Ok([ret.into(), 0]).into();
```

**Pull request merged**